demos/node-classification/deep-graph-info-max/gcn-deep-graph-info-max-cora-example.ipynb

```diff
@@ -0,0 +1,284 @@
+      "text/plain": [
+       "<Figure size 504x288 with 1 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "es = EarlyStopping(monitor=\"loss\", min_delta=0, patience=20)\n",
+    "history = model.fit(gen, epochs=1000, verbose=0, callbacks=[es])\n",
+    "plot_history(history)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Extracting Embeddings and Logistic Regression\n",
+    "\n",
+    "We create an embedding model using `GCNInfoMax.embedding_model` to obtain the node embeddings. Then we use logistic regression to predict which class the node belongs to."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "x_emb_in, x_emb_out = infomax.embedding_model(model)\n",
+    "emb_model = Model(inputs=x_emb_in, outputs=x_emb_out)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Test classification accuracy: 0.7551271534044298\n"
+     ]
+    }
+   ],
+   "source": [
+    "train_subjects, test_subjects = model_selection.train_test_split(\n",
+    "    node_subjects, train_size=0.1, test_size=None, stratify=node_subjects\n",
+    ")\n",
+    "\n",
+    "test_gen = generator.flow(test_subjects.index)\n",
+    "train_gen = generator.flow(train_subjects.index)\n",
+    "\n",
+    "test_embeddings = emb_model.predict(test_gen)\n",
+    "train_embeddings = emb_model.predict(train_gen)\n",
+    "\n",
+    "lr = LogisticRegression(multi_class=\"auto\", solver=\"lbfgs\")\n",
+    "lr.fit(train_embeddings, train_subjects)\n",
+    "\n",
+    "y_pred = lr.predict(test_embeddings)\n",
+    "print(\"Test classification accuracy:\", (y_pred == test_subjects).mean())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Visualisation with TSNE\n",
+    "\n",
+    "Here we visualize the node embeddings with TSNE. As you can see below, the Deep Graph Infomax model produces well separated embeddings using unsupervised training."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "all_embeddings = emb_model.predict(generator.flow(G.nodes()))\n",
+    "\n",
+    "y = node_subjects.astype(\"category\").cat.codes\n",
```

Comment:
A more natural phrasing of this (as in, more applicable for other tasks a user might do with this) might be to leave this as categorical with the human readable names, and do the `cat.codes` below:

```python
y = node_subjects.astype("category")

...

ax.scatter(
    ...
    c=emb_transformed["label"].cat.codes,
    ...
)
```

Alternatively, leave this line as is, but remove the `astype("category")` from the plotting cell, since it's not needed.